lib/src/rocket.rs

```diff
@@ -91,8 +93,18 @@ impl hyper::Handler for Rocket {
 #[cfg(not(feature = "tls"))]
 macro_rules! serve {
     ($rocket:expr, $addr:expr, |$server:ident, $proto:ident| $continue:expr) => ({
-        let ($proto, $server) = ("http://", hyper::Server::http($addr));
-        $continue
+        let $proto = "http://";
+        if $addr.starts_with("unix://") {
+            let path = &$addr[7..];
```

Comment:
This should be determined at configuration parsing time. See my comments in the config module.